### ISSUE TITLE
Automation Parameters

### ISSUE THREAD

**Issue by ortegaantunes** [nemerosa/ontrack-jenkins-cli-pipeline#17]
Hi @dcoraboeuf 

Is possible to have more parameters linked to properties on the Build and Validation Stamp levels. The idea is to use these fields to fill with information from API of other services that yet do not have integration.

Parameters required:

Build
- Messages
- Links
- Meta Information****

Validation Stamp

- Messages
- Links
- Meta Information****

****Not required, but nice to have

![Screenshot 2021-09-07 at 12 09 28](https://user-images.githubusercontent.com/29414973/132327316-880787e2-3bb2-4939-ad1d-6e237a9bfa16.png)

**Comment by dcoraboeuf** [comment]
Builds:

* message - will do
* links - will do, depends on https://github.com/nemerosa/ontrack/issues/904
* meta information - will do, depends on https://github.com/nemerosa/ontrack/issues/905

Validation stamp (or do you mean a validation run, ie the association between a build and a validation stamp?)

* message - will do
* links - not applicable? Or can you explain?
* meta information - will do, depends on https://github.com/nemerosa/ontrack/issues/905

**Comment by dcoraboeuf** [comment]
Besides the absence of some mutations in the GraphQL API, you can use the existing ones in your pipeline using the [`ontrackCliGraphQL`](https://github.com/nemerosa/ontrack-jenkins-cli-pipeline/blob/main/vars/ontrackCliGraphQL.md) step - this will unblock some of the items

**Comment by dcoraboeuf** [comment]
Ontrack 4.0.35 contains the new mutations - #904 & #905

I'll soon work on adding the proper signatures for the builds & validation runs.

**Comment by dcoraboeuf** [comment]
- [ ] message on create build
- [ ] message on existing build
- [ ] links on existing build
- [ ] meta info on create build
- [ ] meta info on existing build
- [ ] message on create validation run
- [ ] meta info on create validation run

**Comment by ortegaantunes** [comment]
> Builds:
> 
> * message - will do
> * links - will do, depends on [GraphQL mutation to create links between builds ontrack#904](https://github.com/nemerosa/ontrack/issues/904)
> * meta information - will do, depends on [GraphQL mutation to set meta information properties ontrack#905](https://github.com/nemerosa/ontrack/issues/905)
> 
> Validation stamp (or do you mean a validation run, ie the association between a build and a validation stamp?)
> 
> * message - will do
> * links - not applicable? Or can you explain?
> * meta information - will do, depends on [GraphQL mutation to set meta information properties ontrack#905](https://github.com/nemerosa/ontrack/issues/905)

Hi @dcoraboeuf I noticed your question just now =/. Yes, I meant a validation run.

**Comment by dcoraboeuf** [comment]
Available in 1.0.3 (v1)

Note that this relies on Ontrack 4.0.39 (minimum version)
